openc3/lib/openc3/utilities/bucket_file_cache.rb

```diff
@@ -68,12 +68,30 @@ def initialize(bucket_path)
     @topic_prefix = "#{scope}__#{type}__{#{target_name}}"
   end
 
+  # Returns true if the file was retrieved and added to the disk
+  # Returns false if the file already exists
+  # Raises an error on retrieval errors
   def retrieve(client = @bucket, uncompress = true)
     @mutex.synchronize do
       local_path = "#{BucketFileCache.instance.cache_dir}/#{File.basename(@bucket_path)}"
       unless File.exist?(local_path)
         OpenC3::Logger.debug "Retrieving #{@bucket_path} from logs bucket"
-        client.get_object(bucket: ENV['OPENC3_LOGS_BUCKET'], key: @bucket_path, path: local_path)
+
+        retry_count = 0
+        begin
+          client_result = client.get_object(bucket: ENV['OPENC3_LOGS_BUCKET'], key: @bucket_path, path: local_path)
+          unless File.exist?(local_path)
+            raise "Local file does not exist after get_object: #{client_result.inspect}"
+          end
+        rescue => err
+          # Try to retrieve the file three times
```

Comment:
Do you think this will address real world issues? We're asking for a file with get_object and for some reason it fails so retry and it works?